### PR TITLE
fix(exp-improve): fixes for exp feature

### PR DESCRIPTION
**Pull request by dmitryuck**

### CODE REVIEW

File: src/about/about.component.ts [hunk 1]
```diff
@@ -48,6 +52,22 @@ export class AboutComponent implements AfterViewInit, OnDestroy {
       this.about = val.data;
 
       this.aboutContent = this.languageService.getSunitizedString(this.about.context);
+
+      setTimeout(() => {
```
Comment (korel-san) [inline src/about/about.component.ts]: @mdmitriy I don't like it. Explain, pls, what happens here and why you used setTimeout?

File: src/common/income-calc/income-calc.service.ts [hunk 2]
```diff
@@ -27,18 +34,28 @@ export class IncomeCalcService {
       }
     }
 
-    return resultIncome * currencyValue;
+    let currencyIncome = resultIncome * currencyValue;
+
+    return (currencyIncome >= 10 ? this.math.round(currencyIncome) : this.math.round(currencyIncome, true)).replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1 ');
```
Comment (korel-san) [inline src/common/income-calc/income-calc.service.ts]: @mdmitriy this.math.round(currencyIncome, currencyIncome < 10) && string type

File: src/common/income-calc/income-calc.service.ts [hunk 3]
```diff
@@ -27,18 +34,28 @@ export class IncomeCalcService {
...
   }
 
   public getTimeUnitByCode(units: any[], code: string): any {
-    return units.find(unit => unit.code === code);
+    if (units) {
```
Comment (korel-san) [inline src/common/income-calc/income-calc.service.ts]: @mdmitriy remove it, but add default value or call this function only with arrays (no undefined values)

File: src/common/math/math.service.ts [hunk 4]
```diff
@@ -2,13 +2,11 @@ import { Injectable } from '@angular/core';
 
 @Injectable()
 export class MathService {
-  public round(income: number): string | number {
+  public round(income: number, dec: boolean = false): string | number {
```
Comment (korel-san) [inline src/common/math/math.service.ts]: @mdmitriy incomeRound, income < 10

File: src/common/math/math.service.ts [hunk 5]
```diff
@@ -2,13 +2,11 @@ import { Injectable } from '@angular/core';
...
     if (!income) {
       return 0;
     }
 
-    let roundIncome: number = Math.round(income);
-
-    return roundIncome.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1 ');
+    return dec ? parseFloat(income.toString()).toFixed(2) : Math.round(income).toString();
```
Comment (korel-san) [inline src/common/math/math.service.ts]: @mdmitriy income.toFixed(2)

File: src/matrix/matrix.component.ts [hunk 6]
```diff
@@ -222,10 +221,6 @@ export class MatrixComponent implements OnDestroy, AfterViewInit {
           this.isPreviewView = false;
 
           //this.showHideHeader(false);
```
Comment (korel-san) [inline src/matrix/matrix.component.ts]: @mdmitriy remove

File: src/matrix/matrix.component.ts [hunk 7]
```diff
@@ -442,7 +431,7 @@ export class MatrixComponent implements OnDestroy, AfterViewInit {
     this.hoverPinnedPlace.emit(place);
   }
 
-  public showHideHeader(mode: boolean): void {
+  /*public showHideHeader(mode: boolean): void {
```
Comment (korel-san) [inline src/matrix/matrix.component.ts]: @mdmitriy remove

File: src/shared/header/header.component.ts [hunk 8]
```diff
@@ -509,7 +467,9 @@ export class HeaderComponent implements OnDestroy, AfterViewInit, OnInit {
 
     let incomeDesktopFilterElement = this.element.querySelector('.filter') as HTMLElement;
 
-    incomeDesktopFilterElement.classList.remove('opened');
+    if (incomeDesktopFilterElement) {
```
Comment (korel-san) [inline src/shared/header/header.component.ts]: @mdmitriy via class in template?

File: src/shared/street-filter/street-filter.service.ts [hunk 9]
```diff
@@ -83,7 +84,7 @@ export class StreetFilterDrawService {
       .enter()
       .append('text')
       .text((d: any) => {
-        return this.math.round(d) + '$';
+        return this.math.round(d * this.currencyUnit.value) + this.currencyUnit.symbol;
```
Comment (korel-san) [inline src/shared/street-filter/street-filter.service.ts]: @mdmitriy ``

File: src/shared/street-filter/street-filter.service.ts [hunk 10]
```diff
@@ -591,11 +592,11 @@ export class StreetFilterDrawService {
...
 
     this.leftScrollText
-      .text(`$${incomeL}`)
+      .text(`${this.currencyUnit.symbol}${incomeL}`)
       .attr('x', ()=> xL + this.halfOfStreetOffset - 5.5 - leftScrollTextWidth / 2);
 
     this.rightScrollText
-      .text(`$${incomeR}`)
+      .text(`${this.currencyUnit.symbol}${incomeR}`)
       .attr('x', ()=> xR + this.halfOfStreetOffset + 5.5 - rightScrollTextWidth / 2);
```
Comment (korel-san) [inline src/shared/street-filter/street-filter.service.ts:600]: @mdmitriy function for formula and for text

**Review by korel-san — CHANGES_REQUESTED** [review]

File: src/shared/street-filter/street-filter.service.ts (hunk 10 above)
Comment (dmitryuck) [reply src/shared/street-filter/street-filter.service.ts:600]: there are two different input arguments xL and leftScrollTextWidth. no reason to move this line to new function

File: src/about/about.component.ts (hunk 1 above)
Comment (dmitryuck) [reply src/about/about.component.ts]: setTimeout need to run fragment of code asyn, delaying is not neccessarry

**Pull request merged**